elasticsearch_dsl/search_base.py

```diff
@@ -714,10 +714,10 @@ def ensure_strings(
                 Dict[str, List[Union[str, "InstrumentedField"]]],
             ]
         ) -> Union[str, List[str], Dict[str, List[str]]]:
-            if isinstance(fields, list):
-                return [str(f) for f in fields]
-            elif isinstance(fields, dict):
+            if isinstance(fields, dict):
                 return {k: ensure_strings(v) for k, v in fields.items()}
+            elif not isinstance(fields, (str, InstrumentedField)):
+                return [str(f) for f in fields]
```

Comment:
Have you considered a more explicit check, like explicitly allowing lists and tuples or looking for for `collections.abc.Iterable`?

Reply:
I'll add a comment for now. Basically the way I'm thinking about this is that in this case the typing change broke how the library works at runtime. My goal was to address the runtime problem. As I mentioned back when we did the typing changes, I think we could replace specific types such as List and Dict with more generic ones in a lot of places, but when I look at that I'll do it for the whole library.